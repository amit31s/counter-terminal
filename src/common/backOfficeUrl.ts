import { STORAGE_KEYS } from "@ct/common/enums";
import { getItem } from "@ct/utils";
import { envProvider } from "./platformHelper";

export enum BackOfficePageEnum {
  taskManagement = "taskManagement",
  pouchPreperation = "pouchPreperation",
}

export const generateBackOfficeURL = () => {
  const currentPage = getItem(STORAGE_KEYS.CTSTK0003);
  const deviceFadcode = getItem(STORAGE_KEYS.CTSTK0004);
  return `${envProvider.REACT_APP_BACK_OFFICE_URL}/${currentPage}?navigatedFromCT=true&fadcode=${deviceFadcode}`;
};
