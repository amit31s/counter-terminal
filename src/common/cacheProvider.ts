import Cache from "@aws-amplify/cache";
import { CacheItemOptions as AmplifyCacheInterface } from "@aws-amplify/cache/src/types/Cache";

Cache.configure({
  itemMaxSize: 400000, // 400kb
});

type CacheValue = string | number | boolean | Record<string, unknown> | unknown[];

const setItem = (key: string, value: CacheValue, options?: AmplifyCacheInterface): void =>
  Cache.setItem(key, value, options);
const getItem = (key: string, options?: AmplifyCacheInterface): CacheValue | unknown =>
  Cache.getItem(key, options);
const removeItem = (key: string): void => Cache.removeItem(key);
const clear = (): void => Cache.clear();

export default Object.freeze({
  setItem,
  getItem,
  removeItem,
  clear,
});
