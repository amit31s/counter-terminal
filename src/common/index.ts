export * from "./backendUrl";
export * from "./backOfficeUrl";
export * from "./components";
export * from "./getProduct";
export * from "./helpers";
export * from "./hooks";
export * from "./selectors";
export * from "./state";
export * from "./theme";
