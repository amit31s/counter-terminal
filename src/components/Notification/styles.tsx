import FontFamily from "@ct/utils/Scaling/FontFamily";
import { StyleSheet } from "react-native";

export default StyleSheet.create({
  notificationHolder: {
    borderWidth: 1,
    borderRadius: 4,
    flex: 1,
    flexDirection: "row",
    justifyContent: "center",
  },
  warningIconHolder: {
    flex: 0.7,
  },
  warningIcon: {
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    flex: 1,
  },
  textHolder: {
    flex: 10,
    margin: "3%",
  },
  titleText: {
    fontFamily: FontFamily.FONT_NUNITO_BOLD,
    fontSize: 24,
    textAlign: "left",
  },
  bodyText: {
    fontFamily: FontFamily.FONT_NUNITO_REGULAR,
    fontSize: 24,
  },
  closeIconHolder: {
    flexDirection: "row",
    justifyContent: "center",
    marginTop: "2%",
    flex: 1,
  },
});
