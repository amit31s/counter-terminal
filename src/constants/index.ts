import AppConstants from "./AppConstants";
import colorConstants from "./ColorConstants";
import ErorrConstants from "./ErrorConstants";
import imageConstants from "./ImageConstants";
import stringConstants from "./StringsConstants";
import Styles from "./styles";
import { GetAccessAndIdTokenURL, getBaseURL, URLConstants } from "./URLConstants";
export { APP_CONSTANTS } from "./AppConstants";
export * from "./Button";
export { COLOR_CONSTANTS } from "./ColorConstants";
export * from "./featureFlags";
export * from "./initialWindowMetrics";
export * from "./queryKeys";
export * from "./screenEnums";
export { SELF_REGISTER_URLS } from "./selfRegisterURLs";
export * from "./StateConstant";
export { STRING_CONSTANTS } from "./StringsConstants";
export * from "./Text";
export {
  Styles,
  colorConstants,
  stringConstants,
  URLConstants,
  AppConstants,
  ErorrConstants,
  imageConstants,
  getBaseURL,
  GetAccessAndIdTokenURL,
};
