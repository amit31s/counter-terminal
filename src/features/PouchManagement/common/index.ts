export * from "./SubmitPouchButton";
export * from "./UnableToScan";
