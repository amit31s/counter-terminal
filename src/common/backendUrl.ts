/* istanbul ignore file */

import { featureFlags } from "@ct/constants";
import isElectron from "is-electron";
import { envProvider } from "./platformHelper";

const forgeRockRedirectUrl =
  envProvider.NODE_ENV === "development" || isElectron()
    ? envProvider.REACT_APP_FORGEROCK_COGNITO_REDIRECT_ELECTRON
    : envProvider.REACT_APP_FORGEROCK_COGNITO_REDIRECT_SIM;

// should contain backend urls for different environments to make switching easier
export const SERVER_ROOT: string = `https://api.spm-${envProvider.REACT_APP_ENV}.com` || "";

export const AWS_COGNITO_REGION: string = envProvider.REACT_APP_AWS_COGNITO_REGION || "";
export const AWS_COGNITO_USER_POOL_ID: string =
  envProvider.REACT_APP_AWS_COGNITO_USER_POOL_ID || "";
export const AWS_COGNITO_WEB_CLIENT_ID: string =
  envProvider.REACT_APP_AWS_COGNITO_WEB_CLIENT_ID || "";

export const FORGEROCK_COGNITO_REGION: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_REGION || "";
export const FORGEROCK_COGNITO_USER_POOL_ID: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_USER_POOL_ID || "";
export const FORGEROCK_COGNITO_WEB_CLIENT_ID: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID || "";
export const FORGEROCK_COGNITO_OAUTH_DOMAIN: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_OAUTH_DOMAIN || "";
export const FORGEROCK_COGNITO_S3_BUCKET: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_S3_BUCKET || "";
export const FORGEROCK_COGNITO_REDIRECT: string = forgeRockRedirectUrl || "";

const federationManagerUrlLocal = "http://localhost:4000/apps/federation-manager";
const federationManagerUrlDeployed =
  (envProvider.REACT_APP_USING_ELECTRON === "true"
    ? envProvider.REACT_APP_FEDERATION_MANAGER_URL_STANDALONE
    : envProvider.REACT_APP_FEDERATION_MANAGER_URL_SIMULATOR) || "";

export const FEDERATION_MANAGER_URL: string = featureFlags?.useLocalFederationManager
  ? federationManagerUrlLocal
  : federationManagerUrlDeployed;

export const POL_DEVICE_SERVER_HOST: string = envProvider.REACT_APP_POL_DEVICE_SERVER_HOST || "";
export const POL_DEVICE_SERVER_SIMULATED: boolean =
  envProvider.REACT_APP_USING_ELECTRON === "true"
    ? envProvider.REACT_APP_POL_DEVICE_SERVER_SIMULATED_ELECTRON === "true"
    : envProvider.REACT_APP_POL_DEVICE_SERVER_SIMULATED_SIMULATOR === "true";
export const CASH_TRANSFER_OUT_DISABLED: boolean =
  envProvider.REACT_APP_CASH_TRANSFER_OUT_DIASBLED === "true" || false;
export const POUCH_ACCEPTANCE_DISABLED: boolean =
  envProvider.REACT_APP_POUCH_ACCEPTANCE_DISABLED === "true" || false;
export const LOGOUT_CASHDRAWER_DISABLED: boolean =
  envProvider.REACT_APP_LOGOUT_CASHDRAWER_DISABLED === "true" || false;
export const POUCH_DISPATCH_DISABLED: boolean =
  envProvider.REACT_APP_POUCH_DISPATCH_DISABLED === "true" || false;
export const PMS_VISIBLE: boolean = envProvider.REACT_APP_PMS_VISIBLE === "true" || false;
export const QUANTITY_BUTTON_DISABLED: boolean =
  envProvider.REACT_APP_QUANTITY_BUTTON_DISABLED === "true" || false;
export const INITIALISE_PED_ON_LOGIN: boolean =
  envProvider.REACT_APP_INITIALISE_PED_ON_LOGIN === "true" || false;
// allows for local override of tpv for pin pad
export const POL_PED_TPV_OVERRIDE: string = envProvider.REACT_APP_POL_PED_TPV_OVERRIDE || "";
export const MSR_ENABLED: boolean = envProvider.REACT_APP_MSR_ENABLED === "true" || false;
export const REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID =
  envProvider.REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID;
