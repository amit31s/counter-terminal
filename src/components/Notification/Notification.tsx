import { StyledCloseIcon } from "@ct/assets/icons";
import { COLOR_CONSTANTS, TEXT } from "@ct/constants";
import { Box, Pressable, Text } from "native-base";
import { ReactChild } from "react";
import styles from "./styles";

export enum NotificationStatus {
  WARNING = "warning",
  FAILURE = "failure",
  SUCCESS = "success",
}

type NotificationType = {
  icon?: ReactChild;
  onClosePress: () => void;
  title: string;
  message: string;
  id: string;
  type: NotificationStatus;
};

export type BackgroundAndBorderColor = { backgroundColor: string; borderColor: string };
export const backgroundAndBorderColor = (type: NotificationStatus): BackgroundAndBorderColor => {
  const { notificationBackground, notificationBorder } = COLOR_CONSTANTS;
  if (type === NotificationStatus.WARNING) {
    return {
      backgroundColor: notificationBackground,
      borderColor: notificationBorder,
    };
  }
  return {
    backgroundColor: notificationBackground,
    borderColor: notificationBorder,
  };
};

export const Notification = ({
  icon,
  onClosePress,
  title,
  message,
  id,
  type,
}: NotificationType) => {
  const { backgroundColor, borderColor } = backgroundAndBorderColor(type);
  return (
    <Box
      background={backgroundColor}
      borderColor={borderColor}
      style={styles.notificationHolder}
      testID={id}
    >
      {icon && (
        <Box style={styles.warningIconHolder}>
          <Box style={styles.warningIcon}>{icon}</Box>
        </Box>
      )}
      <Box style={styles.textHolder} margin={"3"}>
        <Text style={styles.titleText}>{title}</Text>
        <Text style={styles.bodyText}>{message}</Text>
      </Box>
      <Box style={styles.closeIconHolder}>
        <Pressable testID={TEXT.CTTXT00011} onPress={onClosePress}>
          <StyledCloseIcon />
        </Pressable>
      </Box>
    </Box>
  );
};
