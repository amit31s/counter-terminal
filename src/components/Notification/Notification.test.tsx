import { render, setupUser } from "@ct/common";
import { TEXT, colorConstants } from "@ct/constants";
import { Notification, NotificationStatus, backgroundAndBorderColor } from "./Notification";

const onClosePress = jest.fn();

describe("notification component", () => {
  describe("background and border", () => {
    it("should render correct background and border color", () => {
      const statusTypes = Object.values(NotificationStatus);
      statusTypes.forEach((statusType) => {
        const { notificationBackground, notificationBorder } = colorConstants;
        expect(backgroundAndBorderColor(statusType)).toEqual({
          backgroundColor: notificationBackground,
          borderColor: notificationBorder,
        });
      });
    });
  });

  describe("component holder", () => {
    it("renders with correct styling", () => {
      const { getByTestId } = render(
        <Notification
          onClosePress={onClosePress}
          title="Title"
          message="Body"
          id={TEXT.CTTXT00010}
          type={NotificationStatus.WARNING}
        />,
      );
      const componentHolder = getByTestId(TEXT.CTTXT00010);
      expect(componentHolder).toMatchSnapshot();
    });
  });
  describe("title text", () => {
    it("renders correctly", () => {
      const { getByText } = render(
        <Notification
          onClosePress={onClosePress}
          title="Title"
          message="Body"
          id={TEXT.CTTXT00010}
          type={NotificationStatus.WARNING}
        />,
      );
      const titleText = getByText("Title");
      expect(titleText).toBeTruthy();
    });

    it("has the correct styling", () => {
      const { getByText } = render(
        <Notification
          onClosePress={onClosePress}
          title="Title"
          message="Body"
          id={TEXT.CTTXT00010}
          type={NotificationStatus.WARNING}
        />,
      );
      const title = getByText("Title");
      expect(title).toMatchSnapshot();
    });
  });

  describe("body text", () => {
    it("renders correctly", () => {
      const { getByText } = render(
        <Notification
          onClosePress={onClosePress}
          title="Title"
          message="Body"
          id={TEXT.CTTXT00010}
          type={NotificationStatus.WARNING}
        />,
      );
      const bodyText = getByText("Body");
      expect(bodyText).toBeTruthy();
    });

    it("has the correct styling", () => {
      const { getByText } = render(
        <Notification
          onClosePress={onClosePress}
          title="Title"
          message="Body"
          id={TEXT.CTTXT00010}
          type={NotificationStatus.WARNING}
        />,
      );
      const bodyText = getByText("Body");
      expect(bodyText).toMatchSnapshot();
    });
  });

  describe("close button", () => {
    it("fires onClosePress once on click", async () => {
      const user = setupUser();
      const { getByTestId } = render(
        <Notification
          onClosePress={onClosePress}
          title="Title"
          message="Body"
          id={TEXT.CTTXT00010}
          type={NotificationStatus.WARNING}
        />,
      );
      const closeButton = getByTestId(TEXT.CTTXT00011);
      await user.click(closeButton);
      expect(onClosePress).toBeCalledTimes(1);
    });
  });
});
