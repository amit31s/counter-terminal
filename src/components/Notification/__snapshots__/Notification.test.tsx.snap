// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notification component body text has the correct styling 1`] = `
<div
  class="emotion-0 r-color-1njgeiw r-fontFamily-17bz10c r-fontSize-1x35g6 r-fontStyle-13uqrnb r-fontWeight-16dba41 r-letterSpacing-oxtfae r-lineHeight-dhbnww"
  dir="auto"
>
  Body
</div>
`;

exports[`notification component component holder renders with correct styling 1`] = `
.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: #293462;
  width: 32px;
  height: 32px;
}

<div
  class="emotion-0 r-backgroundColor-1lnj8qs r-borderColor-mzlcs0 r-borderRadius-z2wwpe r-borderWidth-rs99b7 r-flex-13awgt0 r-flexDirection-18u37iz r-justifyContent-1777fci"
  data-testid="Suspend Basket WarningMessages TestId"
>
  <div
    class="emotion-0 r-flex-1xgprpv r-margin-n96eu3 r-marginBottom-6gpygo r-marginLeft-19u6a5r r-marginRight-1b7u577 r-marginTop-1s2bzr4"
  >
    <div
      class="emotion-2 r-color-1njgeiw r-fontFamily-p6yezw r-fontSize-1x35g6 r-fontStyle-13uqrnb r-fontWeight-16dba41 r-letterSpacing-oxtfae r-lineHeight-dhbnww r-textAlign-fdjqy7"
      dir="auto"
    >
      Title
    </div>
    <div
      class="emotion-2 r-color-1njgeiw r-fontFamily-17bz10c r-fontSize-1x35g6 r-fontStyle-13uqrnb r-fontWeight-16dba41 r-letterSpacing-oxtfae r-lineHeight-dhbnww"
      dir="auto"
    >
      Body
    </div>
  </div>
  <div
    class="emotion-0 r-flex-13awgt0 r-flexDirection-18u37iz r-justifyContent-1777fci r-marginTop-1743inv"
  >
    <div
      class="emotion-0 r-cursor-1loqt21 r-touchAction-1otgn73"
      data-testid="Notification Close Icon TestId"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`notification component title text has the correct styling 1`] = `
<div
  class="emotion-0 r-color-1njgeiw r-fontFamily-p6yezw r-fontSize-1x35g6 r-fontStyle-13uqrnb r-fontWeight-16dba41 r-letterSpacing-oxtfae r-lineHeight-dhbnww r-textAlign-fdjqy7"
  dir="auto"
>
  Title
</div>
`;
